pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -132,21 +142,49 @@ func RunStartBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args
 	}
 	fmt.Fprintf(out, "%s\n", newBuild.Name)
 
+	var (
+		wg      sync.WaitGroup
+		exitErr error
+	)
+
+	// Wait for the build to complete
+	if waitForComplete {
```

Comment:
you need to do this even when specifying follow so that the exit code is reported.